tests/metrics/test_data.py

```diff
@@ -936,3 +953,62 @@ def test_find_doi_citation_data(tmp_path, files_data, expected_result):
     assert result["https_resolvable_doi"] == expected_result["https_resolvable_doi"]
     assert result["publication_date"] == expected_result["publication_date"]
     assert isinstance(result["cited_by_count"], type(expected_result["cited_by_count"]))
+
+
+@pytest.mark.parametrize(
+    "data, expected",
+    [
+        # Test case 1: Positive correlation with boolean values
+        (
+            [
+                {"result": True, "sustainability_correlation": 1},
+                {"result": False, "sustainability_correlation": 1},
+                {"result": True, "sustainability_correlation": 1},
+            ],
+            0.6666666666666666,  # Two "True" values contribute 1 each, one "False" contributes 0
+        ),
+        # Test case 2: Negative correlation with boolean values
+        (
+            [
+                {"result": True, "sustainability_correlation": -1},
+                {"result": False, "sustainability_correlation": -1},
+                {"result": True, "sustainability_correlation": -1},
+            ],
+            0.3333333333333333,  # Two "True" values contribute 0 each, one "False" contributes 1
+        ),
+        # Test case 3: Mixed correlation with boolean values
+        (
+            [
+                {"result": True, "sustainability_correlation": 1},
+                {"result": False, "sustainability_correlation": -1},
+            ],
+            1.0,  # One "True" with positive correlation contributes 1, one "False" with negative correlation contributes 1
+        ),
+        # Test case 4: Single boolean value with positive correlation
+        (
+            [
+                {"result": True, "sustainability_correlation": 1},
+            ],
+            1.0,  # Single "True" value with positive correlation contributes 1
+        ),
+        # Test case 5: Single boolean value with negative correlation
+        (
+            [
+                {"result": False, "sustainability_correlation": -1},
+            ],
+            1.0,  # Single "False" value with negative correlation contributes 1
+        ),
+        # Test case 6: No valid metrics
+        ([], None),  # No metrics, score should be None
+    ],
+)
+def test_compute_sustainability_score(
+    data: List[Dict[str, Union[int, float, bool]]], expected: float
+):
+    result = compute_sustainability_score(data)
```

Comment:
the variable name `data` can probably be improved

Reply:
Thanks, addressed!